### PR TITLE
New obss notifications

### PR DESCRIPTION
**Pull request by upacyxou**
- new obss

### CODE REVIEW

File: src/helpers/proccessBlocksForNotifications.ts [hunk 1]
```diff
@@ -1,28 +1,52 @@
...
-      ],
-    ]
+export function proccessPostsForNotifications(posts: PostAddedEvent[]) {
+  return posts.map(({ args }) => {
+    const feedId = args[0].toNumber()
+    const postId = args[1].toNumber()
+    const post = args[2]
+    const [sender, metadata, timestamp, threadId, replyTo, numberOfComments] =
+      post
+    return {
```
Comment (smyzin) [inline src/helpers/proccessBlocksForNotifications.ts:40]: You can extract this part and return needed data, just reuse the code:

```
parsePost(post/comment)

function parsePost(item) {
  const [sender, metadata, timestamp, threadId, replyTo, numberOfComments] = item

  return {
      metadata: structToCid(metadata),
      numberOfComments: numberOfComments.toNumber(),
      replyTo: replyTo.toNumber(),
      sender,
      threadId: threadId.toNumber(),
      timestamp: timestamp.toNumber(),
      username: generateRandomName(sender),
  }
}
```

**Review by smyzin — APPROVED** [review]

File: src/controllers/token.ts [hunk 2]
```diff
@@ -2,36 +2,42 @@ import { Body, Controller, Ctx, Delete, Get, Post, Query } from 'amala'
...
 export default class TokenController {
   @Get('/')
   async getData(@Query('currentBlockNumber') currentBlockNumber?: number) {
     const currentBlock = await defaultProvider.getBlockNumber()
-    const postsSinceLastCheck = await getEvents(
-      obssContract.filters.FeedPostAdded(),
-      currentBlockNumber || currentBlock,
+    const commentsSinceLastCheck = await getEvents(
+      getFeedsContract.filters.CommentAdded(),
+      currentBlockNumber ? currentBlockNumber + 1 : currentBlock,
```
Comment (T-Damer) [inline src/controllers/token.ts:21]: Can we make a `const` to clarify why we need this?

File: src/helpers/runNotificationListener.ts [hunk 3]
```diff
@@ -22,15 +22,33 @@ const rootFeeds: { [key: number]: string } = env.isProduction
       ...prodFeeds,
     }
 
-obssContract.on(
-  'FeedPostAdded',
+getFeedsContract.on('CommentAdded', async () => {
+  const allTokens = await TokenModel.find()
+  const fcmTokens = allTokens.reduce((acc: string[], { token }) => {
+    if (!apnRegex.test(token)) {
```
Comment (T-Damer) [inline src/helpers/runNotificationListener.ts:28]: Maybe we can fit this in one-line?

Comment (T-Damer) [reply src/helpers/runNotificationListener.ts:28]: Also don't reverse the logic:
```TS
if (apnRegex.text(token) return acc
acc.push(token)
```

**Review by T-Damer — APPROVED** [review]

File: .env.sample [hunk 4]
```diff
@@ -5,4 +5,7 @@ APN_TEAM_ID=XXXXXXXXXX
 BUNDLE_ID=xxx.xxx.xxx
 ETH_RPC=https://rpc-mainnet.maticvigil.com/
 ETH_NETWORK=maticmum
-NODE_ENV=development
\ No newline at end of file
+NODE_ENV=development
+OBSS_STORAGE_CONTRACT_FEEDS=0xba0600230640CB3D9B76b5bBA9A1945EFBB8621A
```
Comment (T-Damer) [inline .env.sample:9]: Let's use `0x0000000000000000000000000000000000000000`

File: package.json [hunk 5]
```diff
@@ -16,7 +16,8 @@
   },
   "dependencies": {
     "@big-whale-labs/constants": "^0.1.74",
```
Comment (T-Damer) [inline package.json:18]: Let's update those 3 bwl packages

**Review by smyzin — APPROVED** [review]

**Pull request merged**